src/sqs/error-handlers.ts

```diff
@@ -86,9 +87,12 @@ const maybeHandleNonRetryableResponseCode = <MessagePayload extends BaseMessageP
 const maybeHandleRateLimitingError = <MessagePayload extends BaseMessagePayload>(error: Error, context: SQSMessageContext<MessagePayload>): ErrorHandlingResult | undefined => {
 	if (error instanceof GithubClientRateLimitingError) {
 		context.log.warn({ error }, `Rate limiting error, retrying`);
-		const delaySec = error.rateLimitReset + RATE_LIMITING_DELAY_BUFFER_SEC - (Date.now() / 1000);
-		return { retryable: true, retryDelaySec: delaySec, isFailure: true };
+		const buffer = BASE_RATE_LIMITING_DELAY_BUFFER_SEC - context.receiveCount * 10;
```

Comment:
Hmm... 1. could we please move 10 to a constant with a descriptive name? 2. could it be negative, should we handle it accordingly?

Reply:
good news, i just pushed that exact change when you made this comment.

will look at negative option now

Reply:
3. why do we need this sophisticated logic? really we are just calculating some buffer which is from 0 to 60 seconds. I think a clarifying comment is needed